### PR TITLE
Remove the other Color type. The idea is to use AzColor for everything.

### PR DESCRIPTION
**Pull request by Adenilson**

### CODE REVIEW

File: src/azure.rs [hunk 1]
```diff
@@ -261,6 +261,16 @@ impl PartialEq for AzColor {
     }
 }
 
+impl AzColor {
+    pub fn new(r: AzFloat, g: AzFloat, b: AzFloat, a: AzFloat) -> AzColor {
+         AzColor { r: r, g: g, b: b, a: a }
+    }
+
+    pub fn hack(color: AzColor) -> AzColor {
```
Comment (mrobinson) [inline src/azure.rs]: I really don't think we should have a method called "hack."

Comment (Adenilson) [inline src/azure.rs]: Yep, I agree. How you like to have it named?

Comment (mrobinson) [inline src/azure.rs]: Why is it necessary? It's unclear why you need to create a new AzColor where it's used.

Comment (Adenilson) [inline src/azure.rs]: Legacy code calls converting the Color to AzColor and vice-versa. I will
try to remove them.

**Comment by Adenilson** [comment]
Yeah, there is a need to make an explicit copy of a immutable AzColor to pass it to C code. I will rename the function to 'copy' as it better describes it use.

File: src/azure.rs [hunk 2]
```diff
@@ -261,6 +261,16 @@ impl PartialEq for AzColor {
     }
 }
 
+impl AzColor {
+    pub fn new(r: AzFloat, g: AzFloat, b: AzFloat, a: AzFloat) -> AzColor {
+         AzColor { r: r, g: g, b: b, a: a }
+    }
+
+    pub fn copy(color: AzColor) -> AzColor {
```
Comment (mrobinson) [inline src/azure.rs]: If you derive from Copy do you really need to implement it as well?

File: src/azure.rs [hunk 3]
```diff
@@ -244,7 +244,7 @@ pub static AZ_ATT_TRANSFER_DISABLE_B: u32 = 2;
 pub static AZ_ATT_TRANSFER_DISABLE_A: u32 = 3;
 
 #[repr(C)]
-#[derive(Clone)]
+#[derive(Copy, Clone)]
```
Comment (mrobinson) [inline src/azure.rs]: Do you need to add Copy since you are simply using clone below?

**Comment by mrobinson** [comment]
This looks good to me.

**Comment by pcwalton** [comment]
I'm not a huge fan of the `Az` prefix leaking everywhere—the `Az` prefix is for the C binding layer only. Having one color type is of course great though. Could you perhaps reexport the `AzColor` as `Color` via `pub use AzColor as Color`?

**Comment by Adenilson** [comment]
Agreed. Please check the updated patch.

**Comment by Adenilson** [comment]
Needs rebase, will close this request and open a new one.

**Pull request closed without merging**